src/Microsoft.AspNetCore.Razor.Language/Legacy/WhiteSpaceRewriter.cs

```diff
@@ -1,41 +1,65 @@
-// Copyright (c) .NET Foundation. All rights reserved.
+﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System.Collections.Generic;
 using System.Linq;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
 
 namespace Microsoft.AspNetCore.Razor.Language.Legacy
 {
-    internal class WhiteSpaceRewriter : MarkupRewriter
+    internal class WhitespaceRewriter : SyntaxRewriter
     {
-        protected override bool CanRewrite(Block block)
+        public override SyntaxNode Visit(SyntaxNode node)
         {
-            return block.Type == BlockKindInternal.Expression && Parent != null;
+            var result = node;
```

Comment:
This isn't really needed. You could use `node` instead becuase the `result` isn't really ever returned.

Reply:
I don't follow. It is used later in `base.Visit(result)`.

Reply:
I'm saying you could just reassign `node`. This is a nitpick anyways, you do you